plugins/catalog/src/components/CatalogPage/CatalogTabs.tsx

```diff
@@ -48,9 +48,20 @@ type Props = {
  */
 export const CatalogTabs = ({ tabs, onChange }: Props) => {
   const filterGroup = useMemo<FilterGroup>(() => {
+    const otherType = 'other';
+    const wellKnownTypes = tabs.map(t => t.id).filter(t => t !== otherType);
+
     return {
       filters: Object.fromEntries(
-        tabs.map(t => [t.id, (entity: Entity) => entity.spec?.type === t.id]),
+        tabs.map(t => [
+          t.id,
+          (entity: Entity) => {
+            return t.id === otherType &&
+              !wellKnownTypes.includes(entity.spec?.type as any)
```

Comment:
Couldn't this be:

```ts
t.id === otherType && (wellKnownTypes.includes(entity.spec?.type) || entity.spec?.type === t.id)
```

Can we also try and get rid of the `any` cast too, maybe by typing `wellKnownTypes` to `EntityType | string` or something?

Reply:
@benjdlambert - I can make the logic look like that but avoiding the use of `any` or casting seems to be challenging in this case because `Entity.spec` is of type `JsonValue` which is made up of primitives etc. If it were an `any` or likely `unknown` itself this would be easier. Open to ideas, but I don't see a way around it.

Reply:
I can do `(entity: Entity) => t.id === otherType && !(wellKnownTypes?.includes(entity.spec?.type as string) || entity.spec?.type === t.id)` though.

Reply:
Ok, this is updated now with using `as string` and removed the `return`.

Reply:
@benjdlambert - Sorry the boolean logic you suggested does not work as intended so I've left it alone. It's not straightforward because it also needs to match on the tab.ids that are being mapped over. I tried a couple iterations and wasn't really any happier with those either.

Reply:
This works:

```ts
(t.id === otherType && !wellKnownTypes.includes(entity.spec?.type as string)) || entity.spec?.type === t.id
```

Can also be cleaned up a bit with a helper function:

```
const isOtherType = (entity: Entity) => !wellKnownTypes.includes(entity.spec?.type as string);
// ...
(t.id === otherType && isOtherType(entity) || entity.spec?.type === t.id
```